include/matx/operators/cross.h

```diff
@@ -0,0 +1,202 @@
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+/////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "matx/core/type_utils.h"
+#include "matx/operators/base_operator.h"
+
+namespace matx
+{
+
+  /**
+   * Returns cross product of two operators when the last dimensions are 2 or 3
+   */
+  namespace detail {
+    template <typename OpA, typename OpB>
+    class CrossOp : public BaseOp<CrossOp<OpA, OpB>>
+    {
+      private:
+        typename detail::base_type_t<OpA> a_;
+        typename detail::base_type_t<OpB> b_;
+
+        static constexpr int32_t out_rank = cuda::std::max(OpA::Rank(), OpB::Rank());
+        static constexpr int32_t min_rank = cuda::std::min(OpA::Rank(), OpB::Rank());
+
+        cuda::std::array<index_t, out_rank> out_dims_;
+
+        //helpers to simplify later checks
+        bool isA2D_ = a_.Size(a_.Rank() - 1) == 2 ? true : false;
+        bool isB2D_ = b_.Size(b_.Rank() - 1) == 2 ? true : false;
+
+      public:
+        using matxop = bool;
+        using value_type = typename OpA::value_type;
+
+        __MATX_INLINE__ std::string str() const { return "cross()"; }
+        __MATX_INLINE__ CrossOp(const OpA &A, const OpB &B) : a_(A), b_(B) {
+          MATX_STATIC_ASSERT_STR(OpA::Rank() >= 1 && OpB::Rank() >= 1, matxInvalidDim, "Operators to cross() must have rank GTE one.");
+
+          //dims other than the last are batched, so count R-->L, beginning one-left of the right-most dim
+          for (int32_t i = 1; i < min_rank; i++) {
+            MATX_ASSERT_STR(a_.Size(a_.Rank() - 1 - i) == b_.Size(b_.Rank() - 1 - i), matxInvalidSize, "A and B tensors must match batch sizes.");
+          }
+
+          MATX_ASSERT_STR(a_.Size(a_.Rank() - 1) == 3 || a_.Size(a_.Rank() - 1) == 2, matxInvalidSize, "Last dimension of A must have size 2 or 3.")
+          MATX_ASSERT_STR(b_.Size(b_.Rank() - 1) == 3 || b_.Size(b_.Rank() - 1) == 2, matxInvalidSize, "Last dimension of B must have size 2 or 3.")
+        
+          //all input vector sizes (3 x 3, 2 x 3, 3 x 2, 2 x 2) will output to a length-3 vector (plus batch dims)
+          out_dims_[0] = 3;
+          for (int32_t i = 1; i < out_rank; i++) {
+            if (i < a_.Rank()){
+              out_dims_[i] = a_.Size(i);
+            }
+            else{
+              out_dims_[i] = b_.Size(i);
+            }
+          }
+        
+          //promote output to 3D
+          if(isA2D_ && isB2D_){
+            out_dims_[out_dims_.size() - 1] = 3;
+          }
+        };
+
+        template <typename... Is>
+        __MATX_INLINE__ __MATX_DEVICE__ __MATX_HOST__ decltype(auto) operator()(Is... indices) const
+        {
+          cuda::std::array idx{indices...};
+          auto idxOut = idx[idx.size() - 1];
+
+          //create references to individual slices for ease of notation
+          cuda::std::array idx0(idx);
```

Comment:
I think this is creating an unnecessary copy in the constructor. Instead these can be created by just using the type and size of `idx`.